JavaScript/a-retry-wrapper.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const retry = (options = {}) => {
+  const {
+    task = null,
+    recovery = null,
+    handleRecovery = false,
+    tryCount = 3,
+  } = options;
+
+  const errors = [];
+  let count = 0;
+
+  return async (...args) => {
+    while (count++ < tryCount) {
+      try {
+        const res = await task(...args);
+        return res;
+      } catch (err) {
+        errors.push(err);
+        if (recovery) {
+          await recovery(...args).catch((err) => {
```

Comment:
If remove the `.catch` method from `recovery` function and use` try-catch` construction, this wrapper will be compatible with synchronous code